python/cuml/cuml/internals/base.pyx

```diff
@@ -851,3 +851,63 @@ class UniversalBase(Base):
                 raise ex
 
             raise ex
+
+    def as_sklearn(self):
+        """
+        Convert the current GPU-accelerated estimator into a scikit-learn estimator.
+        This method imports and builds an equivalent CPU-backed scikit-learn model,
+        transferring all necessary parameters from the GPU representation to the
+        CPU model. After this conversion, the returned object should be a fully
+        compatible scikit-learn estimator, allowing you to use it in standard
+        scikit-learn pipelines and workflows.
+
+        Returns
+        -------
+        sklearn.base.BaseEstimator
+            A scikit-learn compatible estimator instance that mirrors the trained
+            state of the current GPU-accelerated estimator.
+
+        """
+        self.import_cpu_model()
+        self.build_cpu_model()
+        self.gpu_to_cpu()
+        return self._cpu_model
```

Comment:
Should we not return a deep copy here?

Reply:
For my education, why would we want to deepcopy? Mostly asking because in my experience 99% of cases where someone uses `deepcopy` there is something else that we can do instead or just not do it. Mostly Python "just works" without deepcopy'ing, hence my interest

Reply:
If we simply return a reference to the internal model, any modification to one (additional training or something else) would affect the other. This might create a situation in which the CPU and GPU attributes are out of sync in the cuML estimator. Or inversely, the sklearn estimator returned by the function might silently be updated by the cuML estimator.

Reply:
perhaps it should be a parameter, by default most users probably won't care about needing a deep copy, so I wouldn't do it by default, but if a user needs it then they can request it, what do you guys think?